src/components/Segment.jsx

```diff
@@ -0,0 +1,120 @@
+import { React } from 'react';
+import {
+  IconButton,
+  Button,
+  Tr,
+  Td,
+  Text,
+  Flex,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  useDisclosure,
+  Input,
+} from '@chakra-ui/react';
+import { EditIcon } from '@chakra-ui/icons';
+import { PropTypes } from 'prop-types';
+
+function UpdateSegmentPopup(segment, segmentName, distance, onUpdateSegment, onDeleteSegment) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  let editedSegID = segment;
+  let editedSegName = segmentName;
+  let editedSegDist = distance;
+  return (
+    <>
+      <IconButton
+        size="sm"
+        variant="ghost"
+        colorScheme="white"
+        aria-label="Edit Section"
+        icon={<EditIcon />}
+        onClick={onOpen}
+      >
+        opened
+      </IconButton>
+
+      <Modal size="xl" isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Edit Segment</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            <Input
+              placeholder="Edit segment id here"
+              onChange={event => {
+                editedSegID = event.target.value;
+              }}
+            />
+            <Input
+              placeholder="Edit segment name here"
+              onChange={event => {
+                editedSegName = event.target.value;
+              }}
+            />
+            <Input
+              placeholder="Edit segment distance here"
+              onChange={event => {
+                editedSegDist = event.target.value;
+              }}
+            />
```

Comment:
Since this is for editing an existing segment, it would be better to load the segment's current information into the input box, using `value=`, instead of using a placeholder. Same goes for all other edit functionality (sections, title, etc)